waspc/data/Generator/templates/sdk/tsconfig.json

```diff
@@ -8,6 +8,7 @@
       "dom",
       "DOM.Iterable"
     ],
+    "declaration": true,
```

Comment:
Here's the full story for this fix.

When trying the build the web-app, I got:
```
src/router.tsx:28:23 - error TS2339: Property 'to' does not exist on type 'unknown'.

28           path={route.to}
                         ~~

Found 1 error in src/router.tsx:28
```
To inspect further, I opened the generated `web-app` project in VS code and hovered over the import. TypeScript LSP reported:
```
Could not find a declaration file for module 'wasp/client/router'.
'.../todo-typescript/.wasp/out/sdk/wasp/dist/client/router/index.js' implicitly has an 'any' type.

There are types at '../todo-typescript/node_modules/wasp/client/router/index.ts',
but this result could not be resolved when respecting package.json "exports".
The 'wasp' library may need to update its package.json or typings.
```
By reading through https://github.com/microsoft/TypeScript/issues/52363 and [the TSconfig reference](https://www.typescriptlang.org/tsconfig#declaration), I concluded we needed to emit declarations.

Adding the `declaration: true` compile option fixed the issue, but caused different type-related problems (more on this in other comments).